File: app/src/main/java/com/example/muhammadjon/myproject/activities/SplashActivity.java
package com.example.muhammadjon.myproject.activities;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;
import android.preference.PreferenceManager;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

public class SplashActivity extends AppCompatActivity {
    private SharedPreferences preferences;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        preferences= PreferenceManager.getDefaultSharedPreferences(this);
        new Handler().postDelayed(() -> {
            boolean isFirst = preferences.getBoolean("is_first", true);
            if (isFirst) {
                startActivity(new Intent(SplashActivity.this, RegistrActivity.class));
                finish();
            } else {
                startActivity(new Intent(SplashActivity.this, LoginActivity.class));
                finish();
            }
        },1000);
    }
}

File: app/src/main/java/com/example/muhammadjon/myproject/activities/SplashActivity.kt
package com.example.muhammadjon.myproject.activities

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.os.Handler
import android.preference.PreferenceManager
import android.support.v7.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {
    private var preferences: SharedPreferences? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        preferences = PreferenceManager.getDefaultSharedPreferences(this)
        Handler().postDelayed({
            val isFirst = preferences!!.getBoolean("is_first", true)
            if (isFirst) {
                startActivity(Intent(this@SplashActivity, RegistrActivity::class.java))
                finish()
            } else {
                startActivity(Intent(this@SplashActivity, LoginActivity::class.java))
                finish()
            }
        }, 1000)
    }
}
